File: 3.iOS界面开发（进阶篇）/3. Touch/1.事件机制/Event_Me/Event_Me/TestView.m
//
//  TestView.m
//  Event_Me
//
//  Created by 吴狄 on 2017/7/7.
//  Copyright © 2017年 SharePlus. All rights reserved.
//

#import "TestView.h"

@implementation TestView

/*
// Only override drawRect: if you perform custom drawing.
// An empty implementation adversely affects performance during animation.
- (void)drawRect:(CGRect)rect {
    // Drawing code
}
*/


-(void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event{
    
    
    NSLog(@"touch Began %@",self.name);
    [super touchesBegan:touches withEvent:event];
}

-(void)touchesMoved:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event{
    
    NSLog(@"touch Moved %@",self.name);
}

-(void)touchesEnded:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event{
    
    NSLog(@"touch Ended %@",self.name);
    [super touchesEnded:touches withEvent:event];
}
-(UIView *)hitTest:(CGPoint)point withEvent:(UIEvent *)event{
    
    
    
    NSLog(@"hit test from %@", self.name);
    

    
    UIView *hitView= [super hitTest:point withEvent:event];
    
    
    NSLog(@"return hit view %@, self %@", hitView, self);
    
    
    return hitView;
    
}


-(NSString *)description{
    
    
   return  [NSString stringWithFormat:@"[TestView %@]", self.name];
    
}

@end

File: 3.iOS界面开发（进阶篇）/3. Touch/1.事件机制/Event_Me_Swift/Event_Me_Swift/TestView.swift
//
//  TestView.swift
//  Event_Me_Swift
//
//  Created by 吴狄 on 2017/7/7.
//  Copyright © 2017年 SharePlus. All rights reserved.
//

import UIKit

class TestView: UIView {

    /*
    // Only override draw() if you perform custom drawing.
    // An empty implementation adversely affects performance during animation.
    override func draw(_ rect: CGRect) {
        // Drawing code
    }
    */
    
    
    public var name:String = "";
    
    
    override func hitTest(_ point: CGPoint, with event: UIEvent?) -> UIView? {
        
        
        //NSLog("hit test from \(self.name)");
        
        
        let hitView = super.hitTest(point, with: event)
        
        //print("return hit view \(String(describing: hitView)),self: \(self)")
        
        return hitView;

    }
    
    
    override var description: String{
        
         return   "[TestView \(self.name)]";
        
    }

}
